gym/envs/registration.py

```diff
@@ -78,558 +75,191 @@ def parse_env_id(id: str) -> Tuple[Optional[str], str, Optional[int]]:
-                    "entry point package metadata."
-                )
-            # Replace namespace
-            spec.namespace = self._ns
+def find_newest_version(ns: Optional[str], name: str) -> Optional[int]:
+    version: list[int] = [
+        spec_.version
+        for spec_ in registry.values()
+        if spec_.namespace == ns and spec_.name == name and spec_.version is not None
+    ]
+    return max(version, default=None)
 
-        try:
-            # Get all versions of this spec.
-            versions = self.env_specs.versions(spec.namespace, spec.name)
-
-            # We raise an error if the user is attempting to initialize an
-            # unversioned environment when a versioned one already exists.
-            latest_versioned_spec = max(
-                filter(lambda spec: isinstance(spec.version, int), versions),
-                key=lambda spec: cast(int, spec.version),
-                default=None,
-            )
-            unversioned_spec = next(
-                filter(lambda spec: spec.version is None, versions), None
-            )
 
-            # Trying to register an unversioned spec when versioned spec exists
-            if unversioned_spec and spec.version is not None:
-                message = (
-                    "Can't register the versioned environment "
-                    f"`{spec.id}` when the unversioned environment "
-                    f"`{unversioned_spec.id}` of the same name already exists."
-                )
-                raise error.RegistrationError(message)
-            elif latest_versioned_spec and spec.version is None:
-                message = (
-                    f"Can't register the unversioned environment `{spec.id}` "
-                    f"when version `{latest_versioned_spec.version}` "
-                    "of the same name already exists. Note: the default "
-                    "behavior is that the `gym.make` with the unversioned "
-                    "environment will return the latest versioned environment."
-                )
-                raise error.RegistrationError(message)
-        # We might not find this namespace or name in which case
-        # we should continue to register the environment.
-        except (error.NamespaceNotFound, error.NameNotFound):
-            pass
+def load_env_plugins(entry_point: str = "gym.envs") -> None:
+    # Load third-party environments
+    for plugin in metadata.entry_points(group=entry_point):
+        # Python 3.8 doesn't support plugin.module, plugin.attr
+        # So we'll have to try and parse this ourselves
+        try:
+            module, attr = plugin.module, plugin.attr  # type: ignore  ## error: Cannot access member "attr" for type "EntryPoint"
+        except AttributeError:
+            if ":" in plugin.value:
+                module, attr = plugin.value.split(":", maxsplit=1)
+            else:
+                module, attr = plugin.value, None
+        except:
```

Comment:
Would we want to raise a warning here to say there was an issue loading a plugin, otherwise a plugin may not be loaded for unknown reasons?

Reply:
I'd need to be sure about the consequences of that. This code is just copy-paste of the old implementation, so I'll keep this outside the scope of this PR